GoogleCloudPlatform/anthos-samples
# fix: support re-running the GCE-TF scripts to create two ABM clusters in same project

### Fixes #14

#### Description
- The sample for [installing ABM on GCE using Terraform](https://github.com/GoogleCloudPlatform/anthos-samples/tree/master/anthos-bm-gcp-terraform) cannot be run twice against the same project to create to 2 ABM clusters with different sets of VMs
- The reason for this is that the VM names between the subsequent runs will clash

#### Change summary
- Prefix the GCE VM names with the ABM cluster name _(which is an input variable tot he terraform script)_. This makes sure that the new GCE VMs don't have any clashes. 

#### Related PRs/Issues
- Issue #14
- A [documentation CL](http://cl/408093978) has been opened to update the docs to have the cluster prefix wherever the GCE VM is referred to

#### Testing guidelines
- Follow the steps from [this cloud docs](https://cloud.devsite.corp.google.com/anthos/clusters/docs/bare-metal/latest/try/gce-vms-tf) site to get an Anthos on Baremetal cluster running on GCE VMs
- Now  follow the steps from [this Github readme](https://github.com/GoogleCloudPlatform/anthos-samples/blob/fix-double-run/anthos-bm-gcp-terraform/docs/quickstart.md) to create another ABM cluster on GCE in the same project.
  - When creating the second cluster you have to make sure that you don't run the terraform script from the same git repo that you cloned the first time. This is because that repo will have a `terraform state` from the first run you made. This will lead to terraform just trying to update the existing resources
  - You also must change 2 variables in the [`terraform.tfvars`](https://github.com/GoogleCloudPlatform/anthos-samples/blob/fix-double-run/anthos-bm-gcp-terraform/terraform.tfvars.sample) file the second time so that it doesn't conflict with the first one. 
     - `abm_cluster_id` should be different from what you used the first time
     - `anthos_service_account_name ` should be different from the default one used during the first run. You will not see this variable in the given sample `tfvars` file. This is because the default value for it from [`variable.tf` ](https://github.com/GoogleCloudPlatform/anthos-samples/blob/fix-double-run/anthos-bm-gcp-terraform/variables.tf) was used. To use a different value for this variable just add this variable to your new `terraform.tfvars` file
     - All the other fields in the `tarraform.tfvars` file shall remain the same.

Following the above steps should ideally create 6 GCE VMs and 2 ABM clusters!
